File: 2023-2024/sourcecode/mod1/lab4/12JetpackNavigationBasicsJava/app/src/main/java/com/example/a12_jetpacknavigationbasics_java/FragmentB.java
package com.example.a12_jetpacknavigationbasics_java;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a12_jetpacknavigationbasics_java.databinding.FragmentBBinding;

public class FragmentB extends Fragment {

    private FragmentBBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentBBinding.inflate(inflater);

        binding.textView.setText(String.valueOf(
                getArguments() != null ? getArguments().getInt("value") : 0
        ));

        binding.fabB.setOnClickListener(v -> {
            NavDirections action = FragmentBDirections.actionFragmentBToFragmentA();
            Navigation.findNavController(requireView()).navigate(action);


        });

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1/lab2/41DynamicFragmentBasicsKotlin/app/src/main/java/com/example/a41_dynamicfragmentbasics_kotlin/FragmentB.kt
package com.example.a41_dynamicfragmentbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.a41_dynamicfragmentbasics_kotlin.databinding.FragmentABinding
import com.example.a41_dynamicfragmentbasics_kotlin.databinding.FragmentBBinding

class FragmentB : Fragment() {

    private lateinit var binding: FragmentBBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentBBinding.inflate(inflater)
        return binding.root
    }
}